### PR TITLE
Unified Login: Add tracks for login w/ site address - flow and interaction events

### LINKED ISSUE #2721: Unified Login: Add new tracks events
New track event methods added to `LoginAnalyticsTracker.kt` that need to be properly populated.

### PR DESCRIPTION
**Pull request by AmandaRiu**
This PR partially closes #2721 by doing the following:
- Implementing a new unified login tracks class and wrapper mimicking [WPAndroids PR](https://github.com/wordpress-mobile/WordPress-Android/pull/12089). 
- Adds three new track events:
  - `unified_login_step` (`source`: String, `flow`: String, `step`: String)
  - `unified_login_interaction` (`source`: String, `flow`: String, `step`: String, `click`: String)
  - `unified_login_failure` (`source`: String, `flow`: String, `step`: String, `failure`: String)

**NOTE:** _These events have been added to the tracks spreadsheet under a new "Unified Login" tab._

Additional details about this new way of tracking can be found at pbArwn-AP-p2, but here's a summary. The event `unified_login_step` tracks each step in a particular login flow. The flow changes each time a new login type is selected for example login with store creds, magic link, etc. This event contains 3 properties:
- `source`: Origin of login flow. For now ours will always be `default`. 
- `flow`: The current login flow the user is in (magic link, login by password, etc)
- `step`: A single screen view. 

If the login is successful the final step will be `success`. This event does not track failures - there will be a different event for that. Here's a flow diagram of how these events play out:

UL-site-address-flow(1)

This PR also includes a few `unified_login_interaction` events. These are events fired when the user interacts with the app (such as a click event). For more information on these types of events see pbArwn-I6-p2. The following items have these events associated with them:
- help finding site address button tapped
- send 2FA code button tapped
- submit 2FA button tapped
- login with password link tapped
- show help button tapped
- dialog dismissed
- email address picked from hint
- login with site address button tapped
- continue with wordpress.com button tapped
- login with site credentials tapped
- request magic link
- anytime a `continue` button is clicked - which is logged as `submit`
- magic link open email client button tapped
- reset password link clicked

### Example Output
**Login with password**
```
Tracked: unified_login_step, Properties: {"source":"default","flow":"prologue","step":"prologue"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"prologue","step":"prologue","click":"login_with_site_address"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_site_address","step":"start"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_site_address","step":"start","click":"submit"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_site_address","step":"enter_email_address"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"wordpress_com","step":"start","click":"submit"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_magic_link","step":"start"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_magic_link","step":"start","click":"login_with_password"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_magic_link","step":"start","click":"login_with_password"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_password","step":"start"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_password","step":"start","click":"submit"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_password","step":"success"}
```
**Login w/ Magic Link**
```
Tracked: unified_login_step, Properties: {"source":"default","flow":"prologue","step":"prologue"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"prologue","step":"prologue","click":"login_with_site_address"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_site_address","step":"start"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_site_address","step":"start","click":"submit"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_site_address","step":"enter_email_address"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"wordpress_com","step":"start","click":"submit"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_magic_link","step":"start"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_magic_link","step":"start","click":"request_magic_link"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_magic_link","step":"magic_link_requested"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_magic_link","step":"email_opened"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_magic_link","step":"email_opened","click":"open_email_client"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_magic_link","step":"success"}
```

**Login w/ Store Creds + Magic Link**
```
Tracked: unified_login_step, Properties: {"source":"default","flow":"prologue","step":"prologue"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"prologue","step":"prologue","click":"login_with_site_address"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_site_address","step":"start"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_site_address","step":"start","click":"submit"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_site_address","step":"enter_email_address"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"wordpress_com","step":"start","click":"login_with_site_creds"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_store_creds","step":"username_password"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_store_creds","step":"username_password","click":"submit"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_magic_link","step":"start"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_magic_link","step":"start","click":"request_magic_link"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_magic_link","step":"magic_link_requested"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_magic_link","step":"email_opened"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_magic_link","step":"email_opened","click":"open_email_client"}
Tracked: unified_login_step, Properties: {"source":"default","flow":"login_magic_link","step":"success"}
```

**Failure**
```
Tracked: unified_login_failure, Properties: {"source":"default","flow":"login_store_creds","step":"username_password","failure":"The username or password you entered is incorrect"}
```

**Some interactions**
```
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"prologue","step":"prologue","click":"login_with_site_address"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_site_address","step":"start","click":"submit"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"wordpress_com","step":"start","click":"submit"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_magic_link","step":"start","click":"request_magic_link"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_magic_link","step":"email_opened","click":"open_email_client"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"prologue","step":"prologue","click":"login_with_site_address"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_site_address","step":"start","click":"submit"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"wordpress_com","step":"start","click":"login_with_site_creds"}
Tracked: unified_login_interaction, Properties: {"source":"default","flow":"login_store_creds","step":"username_password","click":"submit"}
```

### To Test
Please test the following scenarios:
- site login with email + password
- site login with email + magic link
- site login with store creds + password
- site login with store creds + magic link

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTrackerWrapper.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    /**
+     * A convenience method for logging an error event with some additional meta data.
+     * @param stat The stat to track.
+     * @param errorContext A string providing additional context (if any) about the error.
+     * @param errorType The type of error.
+     * @param errorDescription The error text or other description.
+     */
+    fun track(stat: Stat, errorContext: String, errorType: String, errorDescription: String) {
```
Comment (nbradbury) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTrackerWrapper.kt:25]: @AmandaRiu I'm not seeing this used anywhere. Am I missing something?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/UnifiedLoginTracker.kt [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        SHOW_EMAIL_HINTS("SHOW_EMAIL_HINTS")
+    }
+
+    enum class Click(val value: String) {
+        SUBMIT("submit"),
+        CONTINUE("continue"),
+        DISMISS("dismiss"),
+        CONTINUE_WITH_WORDPRESS_COM("continue_with_wordpress_com"),
+        LOGIN_WITH_SITE_ADDRESS("login_with_site_address"),
+        LOGIN_WITH_GOOGLE("login_with_google"),
```
Comment (nbradbury) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/UnifiedLoginTracker.kt:152]: A number of these enums are unused. Are they for a future PR?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTrackerWrapper.kt (hunk 1 above)
Comment (AmandaRiu) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTrackerWrapper.kt:25]: It'll be used later when I implement full failure tracking

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/UnifiedLoginTracker.kt (hunk 2 above)
Comment (AmandaRiu) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/UnifiedLoginTracker.kt:152]: Yessir 👍

**Review by nbradbury — APPROVED** [review]
Looks good! :shipit:

**Pull request merged**